### PR TITLE
Publish Stories - Backend

### LINKED ISSUE #273: Publish Stories - Backend
Create an endpoint that updates the stories visibility. Takes in a list of stories, and updates database accordingly.

### PR DESCRIPTION
**Pull request by Shehryar21**
closes #273 

The request payload is an array of stories similar to the array of stories returned be get all stories endpoint.

### CODE REVIEW

File: restapi/api.go [hunk 1]
```diff
@@ -63,6 +63,7 @@ func Router(db *gorm.DB, locationFinder location.LocationFinder) (http.Handler,
 		rest.GetHandler(r, "/story/{storyID}", api.ReturnStoryByID)
 		rest.DeleteHandler(r, "/story/{storyID}", api.DeleteStoryByID)
 		rest.PutHandler(r, "/story/{storyID}", api.EditStoryByID)
+		rest.PutHandler(r, "/publish", api.PublishStories)
```
Comment (Shehryar21) [inline restapi/api.go]: Is the endpoint name "/publish" fine or should it be smth else?

File: restapi/stories.go [hunk 2]
```diff
@@ -377,6 +377,20 @@ func (api api) CreateStories(w http.ResponseWriter, r *http.Request) render.Rend
 	return rest.MsgStatusOK("Stories added successfully")
 }
 
+func (api api) PublishStories(w http.ResponseWriter, r *http.Request) render.Renderer {
+	var stories []models.Story
+	if err := json.NewDecoder(r.Body).Decode(&stories); err != nil {
+		return rest.ErrInvalidRequest(api.logger, "Invalid payload", err)
+	}
+	for _, story := range stories {
```
Comment (nimwijetunga) [inline restapi/stories.go:385]: So just to be clear esentially all this is doing is updating each story object right? One thing to consider would be adding a check that only the `is_visible` field has changed but that might be a bit of work

File: restapi/api.go (hunk 1 above)
Comment (nimwijetunga) [reply restapi/api.go]: maybe `/stories/publish` would be better

File: restapi/stories.go [hunk 3]
```diff
@@ -377,6 +377,20 @@ func (api api) CreateStories(w http.ResponseWriter, r *http.Request) render.Rend
 	return rest.MsgStatusOK("Stories added successfully")
 }
 
+func (api api) PublishStories(w http.ResponseWriter, r *http.Request) render.Renderer {
```
Comment (nimwijetunga) [inline restapi/stories.go:380]: Can we add a test for this, particularly test these two cases:

1. Changing a story that was not visible to visible
2. Changing a story that was visible to not visible

**Review by nimwijetunga — CHANGES_REQUESTED** [review]
A few changes but looks good overall

**Review by nimwijetunga — APPROVED** [review]
LGTM

**Review by AbhiPrasad — APPROVED** [review]

**Pull request merged**